src/wallet3/transaction_constructor.hpp

```diff
@@ -2,17 +2,36 @@
 
 #include <cstdint>
 #include <vector>
+#include <memory>
 #include "address.hpp"
+#include "pending_transaction.hpp"
+#include "daemon_comms.hpp"
 
-namespace wallet
+namespace db
 {
-  struct PendingTransaction;
+  class Database;
+}
 
+namespace wallet
+{
   class TransactionConstructor
   {
-    virtual PendingTransaction
+    public: 
+
+    TransactionConstructor(std::shared_ptr<db::Database> database, std::shared_ptr<DaemonComms> dmn ): db(std::move(database)), daemon(std::move(dmn)) {};
+
+    PendingTransaction
     CreateTransaction(
-        const std::vector<std::pair<address, uint64_t>>& recipients, uint64_t feePerKB) const;
+        const std::vector<TransactionRecipient>& recipients, int64_t feePerKB) const;
+    
+    private:
+
+    void SelectInputs(PendingTransaction& ptx) const;
+    void SelectInputsAndFinalise(PendingTransaction& ptx) const;
+    int64_t EstimateFee() const;
```

Comment:
Does this need to be an estimate, or can we just call this `CalculateFee()`?

Reply:
Fee is dependent on output selection cause of transaction size. Output selection needs to know total amount to send including fee. So it is somewhat circular.

Would love for the final fee amount calculable here. But not sure